src/index.ts

```diff
@@ -8,11 +8,13 @@ import { UseCaseContainer } from './view/UseCaseContainer';
 import ReduxAddToCartEffectHandler from './view/store/cart/ReduxAddToCartEffectHandler';
 import { render } from './view';
 import AddToCartAndRefreshUseCase from './useCases/addToCartAndRefresh';
+import InMemoryCartService from './services/inMemoryCartService';
 
 const store = createStore();
 
-const addToCartRepository = new AddToCartInMemoryRepository();
-const listProductsRepository = new ListProductsInMemoryRepository();
+const cartService = new InMemoryCartService();
```

Comment:
Since we have eventually a single Repository for the Cart can we name the `InMemoryCartService` as a ` InMemoryCartRepositoryStrategy` and create a `CartRepository` (implements both `AddToCartRepository` and `ListProductsRepository`) with a `InMemoryCartRepositoryStrategy` (implements `CartRepositoryStrategy`) injected into it? It will reduce the number of repository implementations since we need only one for now.

Reply:
CartService is a standalone unit that can be reused in completely different scenarios. It will encapsulate data mapping and remote api access. Therefore I believe it does not make sense to call it `InMemoryCartRepositoryStrategy`.

Repository is an interface, which is defined by each use-case independently. The way I was thinking about it was that both `AddToCartInMemoryRepository` and `ListProductsInMemoryRepository `are adapters from `CartService` to the corresponding use-case repository interface.

In order to reduce the number of instances in our current case we can rely on structural typing typescript offers.

`
const cartService = new InMemoryCartService();
const addToCartUseCase = new AddToCartUseCase(cartService); // it is possible because of duck typing
const listProductsUseCase = new ListProductsUseCase(cartService); 
`

But I decided to be the most explicit about the solution from the beginning, since we eventually will have to implement adapters anyway.

Reply:
> Therefore I believe it does not make sense to call it InMemoryCartRepositoryStrategy.

Yep. That makes sense if we plan to implement the whole structure I proposed

> But I decided to be the most explicit about the solution from the beginning, since we eventually will have to implement adapters anyway

Ok (but remember the KISS )